### PR TITLE
feat(acms-appm-15): Create New Appointment

### PR DESCRIPTION
**Pull request by MichaelVacca**
JIRA: link to jira ticket
[TICKET](https://champlainsaintlambert.atlassian.net/jira/software/projects/ACMS/boards/61/backlog?selectedIssue=ACMS-15)

## Context:
This PR is related to the functionality to add a new appointment to a customers account 

## Changes
- Back end methods 
- Back end tests
- Front end components and api calls
- Front end tests 

## Before and After UI (Required for UI-impacting PRs)

Before changes:
- no UI

After Changes:
image

## Dev notes (Optional)
- UI tweaks may need to be added later

### CODE REVIEW

**Review by MohitShankar — APPROVED** [review]
Great job, this will help me with next sprint

File: acms-react-app/src/Pages/Admin_Pages/AppointmentNew_Page/Calendar.js [hunk 1]
```diff
@@ -20,13 +21,66 @@ export default function Calendar() {
...
+
+        const appointmentData = {
+            customerId,
+            vehicleId,
+            appointmentDate: selectDate.format("YYYY-MM-DD") + 'T' + selectedTime,
+            services: selectedService,
+            comments,
+            status: 'PENDING'
+        };
+        if (!customerId || !vehicleId) {
```
Comment (CrisWillCarry) [inline acms-react-app/src/Pages/Admin_Pages/AppointmentNew_Page/Calendar.js:39]: Good job on checking for valid input

File: acms-react-app/src/Pages/Admin_Pages/AppointmentNew_Page/Calendar.js [hunk 2]
```diff
@@ -20,13 +21,66 @@ export default function Calendar() {
...
+            console.error("Customer ID and Vehicle ID are required.");
+            console.log("Appointment data:", appointmentData)
+            return;
+        }
+        console.log("Submitting appointment:", appointmentData);
+
+        const url = `http://localhost:8080/api/v1/customers/${customerId}/appointments`;
+
+        fetch(url, {
```
Comment (CrisWillCarry) [inline acms-react-app/src/Pages/Admin_Pages/AppointmentNew_Page/Calendar.js:48]: We have been using axios, make sure to keep it that way, I am currently adding services so we won't have all these requests in the middle of every file.

File: ateliermecanique-ws/src/test/java/com/champlain/ateliermecaniquews/zelenium/FrontEndTesting.java [hunk 3]
```diff
@@ -444,5 +446,49 @@ public void deleteAllCancelledAppointments() {
         $("p").shouldHave(text("APPOINTMENTS"));
     }
 
+    @Test
+    public void addAppointmentToCustomerAccount(){
```
Comment (CrisWillCarry) [inline ateliermecanique-ws/src/test/java/com/champlain/ateliermecaniquews/zelenium/FrontEndTesting.java:450]: Good Job adding front end testing as well

**Review by CrisWillCarry — COMMENTED** [review]
Overall, good coding

**Pull request merged**